### ISSUE TITLE
Issue with adding new list items

### ISSUE THREAD

**Issue by guciek27** [vazco/uniforms#16]
I have an issue with adding new list items when using autosave={true} on the form. It is fine that the form is validated when new item is attached but at the moment the item does not attach on the first click.

Please have a look at the attached gif below that should explain the issue. Keep in mind that I'm clicking twice on the + icon before new items appear:
![kb2fp827mt](https://cloud.githubusercontent.com/assets/1568836/15674301/363f2718-273c-11e6-88cd-77b5e2992a1e.gif)

**Comment by radekmie** [comment]
Thanks for pointing it out. I'll look at it later.

**Comment by radekmie** [comment]
I've tried, but I can't reproduce it. Could you show the code?

**Comment by guciek27** [comment]
This is the simplified version of the code.
```
<AutoForm schema={ userProfileSchema } autosave={ true } onSubmit={ doc => console.log(doc) }>
  <TextField
    error="First name is required"
    name="name.first"
    placeholder="First Name"
  />
  <TextField
    error="Last name is required"
    name="name.last"
    placeholder="Last Name"
  />

  <ListField name="contactEmails" initialCount={1}>
    <ListItemField name="$">
      <TextField name="address" placeholder="Email address" />
    </ListItemField>
  </ListField>
  <ErrorsField />
</AutoForm>
```

**Comment by serkandurusoy** [comment]
Hmm, this may be related to the fact that `contactEmails` is possibly optional with no initial value defined (so its internal expected initial count is 0) but we explicitly provide an initialCount property that is higher

so to summarize, I think this may be an apparent issue with 

* array is optional and we have initial count
* array min count is less than intial count

**Comment by radekmie** [comment]
Hah! It's not that, @serkandurusoy. It's not even a *bug*. If you look carefully, you'll see, that not the *badge* has an `onClick` event, but only the `icon`. It should work in `1.0.0-rc.11`.

**Comment by serkandurusoy** [comment]
ooh! wow, good catch!

**Comment by serkandurusoy** [comment]
@radekmie  that might not have been it afterall. 

If you watch the gif carefully, the first click registers and fires off validation! so it is not a click target problem. the second click then creates the new input

furthermore, this is not a bootstrap-3/4 input but based on the unstyled and your commit changes bootstrap versions, but the unstyled does not seem to have that wrong placement of click handler anyway.

**Comment by radekmie** [comment]
I don't think, that `validate` is being called - those fields, that @guciek27 posted, have *constant* errors. All in all - if this issue is still a problem, please, comment, and I'll reopen this issue.

**Comment by serkandurusoy** [comment]
I tested this. It does not matter about the error being there all along. I can introduce a new error and still reproduce the issue.

1. make sure there is an error, you can create it by clearing a required field
2. click on + for a list (nothing happens)
3. click again and an input is added

it is even more evident when you start with an empty form that has required fields that are empty to begin with.

the first clicks on the + signs don't create an input, but fires off validation (the form is autosave=true if you think it matters)

**Comment by radekmie** [comment]
Okay, I'll dig into it tomorrow.

**Comment by radekmie** [comment]
Problem was in `ListAddField`, not `ListField` - its `initialCount` was undefined, but it should be passed from `ListField`. Fixed in `1.0.0-rc.12` (I hope).

**Comment by serkandurusoy** [comment]
Thank you @radekmie I'll test this as soon as I can and let you know.

Sent from my iPhone



On Jun 1, 2016, at 17:42, "Radosław Miernik" <notifications@github.com>
wrote:

Problem was in ListAddField, not ListField - its initialCount was
undefined, but it should be passed from ListField. Fixed in 1.0.0-rc.12 (I
hope).

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/vazco/uniforms/issues/16#issuecomment-223014518>, or mute
the thread
<https://github.com/notifications/unsubscribe/AEbz3A7FmrC0nb0BpWgCZkh0tJTMsesgks5qHZpWgaJpZM4Iqe7T>
.

**Comment by serkandurusoy** [comment]
this works now! thank you.
